lib/coinbase/address.rb

```diff
@@ -75,10 +75,138 @@ def faucet
+    def staking_balances(asset_id, mode: :default, options: {})
+      context_model = Coinbase.call_api do
+        stake_api.get_staking_context(
+          {
+            asset_id: asset_id,
+            network_id: Coinbase.normalize_network(network_id),
+            address_id: id,
+            options: {
+              mode: mode
+            }.merge(options)
+          }
+        )
+      end.context
+
+      {
+        stakeable_balance: Coinbase::Balance.from_model_and_asset_id(
+          context_model.stakeable_balance,
+          asset_id
+        ).amount,
+        unstakeable_balance: Coinbase::Balance.from_model_and_asset_id(
+          context_model.unstakeable_balance,
+          asset_id
+        ).amount,
+        claimable_balance: Coinbase::Balance.from_model_and_asset_id(
+          context_model.claimable_balance,
+          asset_id
+        ).amount
+      }
+    end
+
+    # Retrieves the stakeable balance for the supplied asset.
+    # @param asset_id [Symbol] The asset to retrieve the stakeable balance for
+    # @param mode [Symbol] The staking mode. Defaults to :default.
+    # @param options [Hash] Additional options for the staking operation
+    # @return [BigDecimal] The stakeable balance
+    def stakeable_balance(asset_id, mode: :default, options: {})
+      staking_balances(asset_id, mode: mode, options: options)[:stakeable_balance]
+    end
+
+    # Retrieves the unstakeable balance for the supplied asset.
+    # @param asset_id [Symbol] The asset to retrieve the unstakeable balance for
+    # @param mode [Symbol] The staking mode. Defaults to :default.
+    # @param options [Hash] Additional options for the staking operation
+    # @return [BigDecimal] The unstakeable balance
+    def unstakeable_balance(asset_id, mode: :default, options: {})
+      staking_balances(asset_id, mode: mode, options: options)[:unstakeable_balance]
+    end
+
+    # Retrieves the claimable balance for the supplied asset.
+    # @param asset_id [Symbol] The asset to retrieve the claimable balance for
+    # @param mode [Symbol] The staking mode. Defaults to :default.
+    # @param options [Hash] Additional options for the staking operation
+    # @return [BigDecimal] The claimable balance
+    def claimable_balance(asset_id, mode: :default, options: {})
+      staking_balances(asset_id, mode: mode, options: options)[:claimable_balance]
+    end
+
+    # Lists the staking rewards for the address.
+    # @param asset_id [Symbol] The asset to retrieve staking rewards for
+    # @param start_time [Time] The start time for the rewards. Defaults to 1 month ago.
+    # @param end_time [Time] The end time for the rewards. Defaults to the current time.
+    # @param format [Symbol] The format to return the rewards in. Defaults to :usd.
+    # @return [Enumerable<Coinbase::StakingReward>] The staking rewards
+    def staking_rewards(asset_id, start_time: DateTime.now.prev_month(1), end_time: DateTime.now, format: :usd)
```

Comment:
maybe use the same lookback default of nodejs i.e. a week ? https://github.com/coinbase/coinbase-sdk-nodejs/blob/e40237144507247c82055e268f97e2b0b7cf97c0/src/coinbase/address.ts#L93

Reply:
Not sure when it changed for NodeJS, but this has always been a month, we also use 10 days everywhere in our docs, we should better align

Reply:
yeah I think it will be nice to update the docs to remove the code around last 10 days and have it be defaulted to whatever we have here.